### PR TITLE
Some ports from Kadas

### PR DESCRIPTION
**Pull request by m-kuhn**
**Code cleanups**

- Hide away *default translator* implementation
- Rubberband loops (not from Kadas)
- Const correctness
- ~Add inline keyword to declarations to match definitions and avoid "redeclared without dllimport" warnings~ Removed again at least for now, see https://isocpp.org/wiki/faq/inline-functions#where-to-put-inline-keyword

**Feature**

- Add the possibility to use an SVG as rubberband marker
- Add [mil](https://en.wikipedia.org/wiki/Milliradian) units

### CODE REVIEW

File: src/core/qgsunittypes.cpp [hunk 1]
```diff
@@ -1256,6 +1256,8 @@ double QgsUnitTypes::fromUnitToUnitFactor( QgsUnitTypes::AngleUnit fromUnit, Qgs
           return 3600;
         case AngleTurn:
           return 1.0 / 360.0;
+        case AngleMil:
```
Comment (nyalldawson) [inline src/core/qgsunittypes.cpp]: These ones will need to be added to existing angle unit tests

**Review by nyalldawson — COMMENTED** [review]
Nice work!

File: src/core/qgsunittypes.cpp (hunk 1 above)
Comment (m-kuhn) [reply src/core/qgsunittypes.cpp]: Done, some expected failures in their current form

File: src/gui/qgsrubberband.cpp [hunk 2]
```diff
@@ -21,6 +21,7 @@
 #include "qgsproject.h"
 #include "qgsrectangle.h"
 #include <QPainter>
+#include <QSvgRenderer>
```
Comment (m-kuhn) [inline src/gui/qgsrubberband.cpp]:
```suggestion
```

File: src/core/qgsunittypes.cpp (hunk 1 above)
Comment (m-kuhn) [reply src/core/qgsunittypes.cpp]: Done, any more tests to do?

**Pull request merged**